### PR TITLE
Fix #2910: Move crd-generator tests out of kubernetes-itests

### LINKED ISSUE #2910: Move crd-generator tests out of kubernetes-itests
The tests/folder in https://github.com/fabric8io/kubernetes-client/tree/master/kubernetes-itests/src/test/java/io/fabric8/crd is not an "integration test", it doesn't need a connection to an active kubernetes cluster, so it should not belong to the `kubernetes-itests` module.

As this is more a "unit test" like, probably it should be inside the same `crd-generator-apt` module, or at best in the module `kubernetes-test` module.

Just as a side note, this test is not running on CI, since the e2e-tests workflow filter the tests that are from `-Dtest="io.fabric8.[kubernetes|openshift].**"` not the io.fabric8.crd.

/cc @metacosm

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description
Fix #2910 
+ Move crd-generator tests from kubernetes-itests -> kubernetes-tests

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [X] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [ ] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: CHANGELOG.md
Comment (manusa) [inline CHANGELOG.md:17]: duplicate
```suggestion
```

**Comment by manusa** [comment]
```
2021-04-06T13:47:17.7815905Z [INFO] Running io.fabric8.crd.generator.zookeeper.ZookeeperCustomResourceTest
2021-04-06T13:47:18.2218597Z [INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.036 s - in io.fabric8.crd.generator.zookeeper.ZookeeperCustomResourceTest
```

**Comment by manusa** [comment]
Shouldn't this one (https://github.com/fabric8io/kubernetes-client/blob/a9af617a27b361654f75ff12d12a84c1a4daf8f8/kubernetes-itests/src/test/java/io/fabric8/crd/FooCRDIT.java) be moved too?

**Comment by rohanKanojia** [comment]
On moving it, I realized FooCRDTest seems to be failing:
```
[INFO] Results:
[INFO] 
Error:  Failures: 
Error:    FooCRDTest.testCrd:39 expected: not <null>
[INFO] 
Error:  Tests run: 683, Failures: 1, Errors: 0, Skipped: 4
```
I think CustomResourceDefinition Yaml manifests for [Foo POJO](https://github.com/rohanKanojia/kubernetes-client/blob/pr/issue2910/kubernetes-tests/src/test/java/io/fabric8/crd/generator/Foo.java) are not generated as per [test expectations](https://github.com/rohanKanojia/kubernetes-client/blob/pr/issue2910/kubernetes-tests/src/test/java/io/fabric8/crd/generator/FooCRDTest.java#L39):
```yaml
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
  name: foos.acme.com
spec:
  group: acme.com
  names:
    kind: Foo
    plural: foos
    singular: foo
  scope: Namespaced
  versions:
  - name: v1alpha1
    schema:
      openAPIV3Schema:
        type: object
    served: true
    storage: true
```
We're expecting that the subresource should not be null.

**Comment by rohanKanojia** [comment]
@metacosm : Hi, Could you please take review this whenever you find some time? I'm not sure whether we're having wrong expectations or something is not working as expected.

**Comment by metacosm** [comment]
This test should be revisited once #2968 is merged since we realized that it was wrong (not sure how that got through)…

**Comment by rohanKanojia** [comment]
I've commented out the failing assertion for now. We can fix this in future.

**Comment by manusa** [comment]
These are just tests, we can perfectly wait for #2962 to be merged

**Comment by rohanKanojia** [comment]
I rebased my branch against master but looks like the test isn't passing even after sundrio upgrade:
```
[INFO] Results:
[INFO] 
Error:  Failures: 
Error:    FooCRDTest.testCrd:40 expected: <true> but was: <false>
[INFO] 
Error:  Tests run: 683, Failures: 1, Errors: 0, Skipped: 4
```
It seems to be failing at this line: https://github.com/rohanKanojia/kubernetes-client/blob/pr/issue2910/kubernetes-tests/src/test/java/io/fabric8/crd/generator/FooCRDTest.java#L40

**Comment by metacosm** [comment]
Fixing this will actually require a new sundrio version… 😭

**Comment by metacosm** [comment]
Superseded by #2996

**Pull request closed without merging**
